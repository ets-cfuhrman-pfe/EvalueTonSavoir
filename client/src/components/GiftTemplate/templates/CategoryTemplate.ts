import { TemplateOptions } from './types';
import QuestionContainer from './QuestionContainerTemplate';
import Title from './TitleTemplate';
import { Category } from 'gift-pegjs';

type CategoryOptions = TemplateOptions & Category;

export default function CategoryTemplate( { title }: CategoryOptions): string {
    return `${QuestionContainer({
        children: Title({
            type: 'Cat√©gorie',
            title: title
        })
    })}`;
}
