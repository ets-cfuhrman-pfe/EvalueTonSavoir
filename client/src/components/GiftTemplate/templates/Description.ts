import { TemplateOptions, Description as DescriptionType } from './types';
import QuestionContainer from './QuestionContainer';
import Title from './Title';
import textType from './TextType';
import { ParagraphStyle } from '../constants';
import { state } from '.';

type DescriptionOptions = TemplateOptions & DescriptionType;

export default function Description({ title, stem }: DescriptionOptions): string {
    return `${QuestionContainer({
        children: [
            Title({
                type: 'Description',
                title: title
            }),
            `<p style="${ParagraphStyle(state.theme)}">${textType({
                text: stem
            })}</p>`
        ]
    })}`;
}
