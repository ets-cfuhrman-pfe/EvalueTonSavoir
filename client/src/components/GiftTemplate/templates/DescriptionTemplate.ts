import { TemplateOptions } from './types';
import QuestionContainer from './QuestionContainerTemplate';
import Title from './TitleTemplate';
import { Description } from 'gift-pegjs';
import StemTemplate from './StemTemplate';

type DescriptionOptions = TemplateOptions & Description;

export default function DescriptionTemplate({ title, formattedStem}: DescriptionOptions): string {
    return `${QuestionContainer({
        children: [
            Title({
                type: 'Description',
                title: title
            }),
            StemTemplate({formattedStem}),
        ]
    })}`;
}
