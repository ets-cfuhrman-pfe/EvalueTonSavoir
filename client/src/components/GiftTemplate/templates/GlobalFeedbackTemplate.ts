import { TemplateOptions } from './types';
import {FormattedTextTemplate} from './TextTypeTemplate';
import { state } from '.';
import { theme } from '../constants';
import { TextFormat } from 'gift-pegjs';

type GlobalFeedbackOptions = TemplateOptions & TextFormat;

export default function GlobalFeedbackTemplate({ format, text }: GlobalFeedbackOptions): string {
    const Container = `
  position: relative;
  margin-top: 1rem;
  padding: 0 1rem;
  background-color: ${theme(state.theme, 'beige100', 'black400')};
  color: ${theme(state.theme, 'beige900', 'gray200')};
  border: ${theme(state.theme, 'beige300', 'black300')} ${state.theme === 'light' ? 1 : 2}px solid;
  border-radius: 6px;
  box-shadow: 0px 2px 5px ${theme(state.theme, 'gray400', 'black800')};
`;

    return (format && text)
        ? `<div style="${Container}">
        <p>${FormattedTextTemplate({format: format, text: text})}</p>
      </div>`
        : ``;
}
