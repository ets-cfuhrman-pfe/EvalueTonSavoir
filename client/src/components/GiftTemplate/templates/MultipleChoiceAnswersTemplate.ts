import { nanoid } from 'nanoid';
import { TemplateOptions } from './types';
import {FormattedTextTemplate} from './TextTypeTemplate';
import AnswerIcon from './AnswerIconTemplate';
import { state } from '.';
import { ParagraphStyle } from '../constants';
import { MultipleChoiceQuestion, TextChoice } from 'gift-pegjs';

type MultipleChoiceAnswerOptions = TemplateOptions & Pick<MultipleChoiceQuestion, 'choices'>;

type AnswerFeedbackOptions = TemplateOptions & Pick<TextChoice, 'formattedFeedback'>;

interface AnswerWeightOptions extends TemplateOptions {
    weight: TextChoice['weight'];
}

export default function MultipleChoiceAnswersTemplate({ choices }: MultipleChoiceAnswerOptions) {
    const id = `id${nanoid(8)}`;

    const isMultipleAnswer = choices.filter(({ isCorrect }) => isCorrect === true).length === 0;

    const prompt = `<span style="${ParagraphStyle(state.theme)}">Choisir une r√©ponse${
        isMultipleAnswer ? ` ou plusieurs` : ``
    }:</span>`;
    const result = choices
        .map(({ weight, isCorrect, formattedText, formattedFeedback }) => {
            const CustomLabel = `
          display: inline-block;
          padding: 0.2em 0 0.2em 0;
        `;

            const inputId = `id${nanoid(6)}`;

            const isPositiveWeight = (weight != undefined) && (weight > 0);
            const isCorrectOption = isMultipleAnswer ? isPositiveWeight : isCorrect;

            return `
        <div class='multiple-choice-answers-container'>
          <input class="gift-input" type="${
              isMultipleAnswer ? 'checkbox' : 'radio'
          }" id="${inputId}" name="${id}">
          ${AnswerWeight({ weight: weight })}
            <label style="${CustomLabel} ${ParagraphStyle(state.theme)}" for="${inputId}">
            ${FormattedTextTemplate(formattedText)}
            </label>
          ${AnswerIcon({ correct: isCorrectOption })}
          ${AnswerFeedback({ formattedFeedback: formattedFeedback })}
          </input>
        </div>
        `;
        })
        .join('');

    return `${prompt}${result}`;
}

function AnswerWeight({ weight }: AnswerWeightOptions): string {
  //   const Container = `
  //     box-shadow: 0px 1px 1px ${theme(state.theme, 'gray400', 'black900')};
  //     border-radius: 3px;
  //     padding-left: 0.2rem;
  //     padding-right: 0.2rem;
  //     padding-top: 0.05rem;
  //     padding-bottom: 0.05rem;
  //   `;

  //   const CorrectWeight = `
  //     color: ${theme(state.theme, 'green700', 'green100')};
  //     background-color: ${theme(state.theme, 'green100', 'greenGray700')};
  //   `;
  //   const IncorrectWeight = `
  //   color: ${theme(state.theme, 'beige600', 'beige100')};
  //   background-color: ${theme(state.theme, 'beige300', 'beigeGray800')};
  // `;

    return weight ? `<span class="answer-weight-container ${weight > 0 ? 'answer-positive-weight' : 'answer-zero-or-less-weight'}">${weight}%</span>` : ``;
        // ? `<span style="${Container} ${
        //       correct ? `${CorrectWeight}` : `${IncorrectWeight}`
        //   }">${weight}%</span>`
        // : ``;
}

function AnswerFeedback({ formattedFeedback }: AnswerFeedbackOptions): string {
    // const Container = `
    //   color: ${theme(state.theme, 'teal700', 'gray700')};
    // `;

    return formattedFeedback ? `<span class="feedback-container">${FormattedTextTemplate(formattedFeedback)}</span>` : ``;
}
