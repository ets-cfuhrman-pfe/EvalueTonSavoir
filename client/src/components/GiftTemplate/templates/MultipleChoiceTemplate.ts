import { TemplateOptions } from './types';
import QuestionContainer from './QuestionContainerTemplate';
import GlobalFeedback from './GlobalFeedbackTemplate';
import Title from './TitleTemplate';
import {textType} from './TextTypeTemplate';
import MultipleChoiceAnswers from './MultipleChoiceAnswersTemplate';
import { ParagraphStyle } from '../constants';
import { state } from '.';
import { MultipleChoiceQuestion } from 'gift-pegjs';

type MultipleChoiceOptions = TemplateOptions & MultipleChoiceQuestion;

export default function MultipleChoiceTemplate({
    title,
    formattedStem,
    choices,
    formattedGlobalFeedback
}: MultipleChoiceOptions): string {
    return `${QuestionContainer({
        children: [
            Title({
                type: 'Choix multiple',
                title: title
            }),
            `<p style="${ParagraphStyle(state.theme)}" class="present-question-stem">${textType(formattedStem)}</p>`,
            MultipleChoiceAnswers({ choices: choices }),
            formattedGlobalFeedback ? GlobalFeedback(formattedGlobalFeedback) : ''
        ]
    })}`;
}
