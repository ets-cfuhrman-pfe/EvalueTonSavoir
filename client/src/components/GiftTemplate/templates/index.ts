import Category from './Category';
import DescriptionTemplate from './Description';
import Essay from './Essay';
import Matching from './Matching';
import MultipleChoice from './MultipleChoice';
import Numerical from './Numerical';
import ShortAnswer from './ShortAnswer';
import TrueFalse from './TrueFalse';
import Error from './Error';
import {
    GIFTQuestion,
    // Category as CategoryType,
    // Description as DescriptionType,
    MultipleChoice as MultipleChoiceType,
    Numerical as NumericalType,
    ShortAnswer as ShortAnswerType,
    // Essay as EssayType,
    TrueFalse as TrueFalseType,
    Matching as MatchingType,
    DisplayOptions
} from './types';

export const state: DisplayOptions = { preview: true, theme: 'light' };

export default function Template(
    { type, ...keys }: GIFTQuestion,
    options?: Partial<DisplayOptions>
): string {
    Object.assign(state, options);

    switch (type) {
        // Category, Description, Essay are not supported?
        // case 'Category':
        //     return Category({ ...(keys as CategoryType) });
        // case 'Description':
        //     return Description({
        //         ...(keys as DescriptionType)
        //     });
        case 'MC':
            return MultipleChoice({
                ...(keys as MultipleChoiceType)
            });
        case 'Numerical':
            return Numerical({ ...(keys as NumericalType) });
        case 'Short':
            return ShortAnswer({
                ...(keys as ShortAnswerType)
            });
        // case 'Essay':
        //     return Essay({ ...(keys as EssayType) });
        case 'TF':
            return TrueFalse({ ...(keys as TrueFalseType) });
        case 'Matching':
            return Matching({ ...(keys as MatchingType) });
        default:
            // TODO: throw error for unsupported question types?
            // throw new Error(`Unsupported question type: ${type}`);
            return ``;
    }
}

export function ErrorTemplate(text: string, options?: Partial<DisplayOptions>): string {
    Object.assign(state, options);

    return Error(text);
}

export {
    Category,
    DescriptionTemplate as Description,
    Essay,
    Matching,
    MultipleChoice,
    Numerical,
    ShortAnswer,
    TrueFalse,
    Error
};
