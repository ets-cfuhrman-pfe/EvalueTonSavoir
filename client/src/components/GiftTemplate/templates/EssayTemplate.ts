import { TemplateOptions } from './types';
import QuestionContainer from './QuestionContainerTemplate';
import Title from './TitleTemplate';
import {textType} from './TextTypeTemplate';
import GlobalFeedbackTemplate from './GlobalFeedbackTemplate';
import { ParagraphStyle, TextAreaStyle } from '../constants';
import { state } from '.';
import { EssayQuestion } from 'gift-pegjs';

type EssayOptions = TemplateOptions & EssayQuestion;

export default function EssayTemplate({ title, formattedStem, formattedGlobalFeedback }: EssayOptions): string {
    return `${QuestionContainer({
        children: [
            Title({
                type: 'Développement',
                title: title
            }),
            `<p style="${ParagraphStyle(state.theme)}">${textType(formattedStem)}</p>`,
            `<textarea class="gift-textarea" style="${TextAreaStyle(
                state.theme
            )}" placeholder="Entrez votre réponse ici..."></textarea>`, 
            (formattedGlobalFeedback && formattedGlobalFeedback.text) ?
                GlobalFeedbackTemplate(formattedGlobalFeedback) : ``
        ]
    })}`;
}
