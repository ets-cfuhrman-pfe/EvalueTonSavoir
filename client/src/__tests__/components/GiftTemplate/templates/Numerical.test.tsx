import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Numerical from '../../../../components/GiftTemplate/templates/Numerical';
import { TemplateOptions, Numerical as NumericalType } from '../../../../components/GiftTemplate/templates/types';

// Mock the nanoid function
jest.mock('nanoid', () => ({
    nanoid: jest.fn(() => 'mocked-id')
  }));

const plainTextMock: TemplateOptions & NumericalType = {
    type: 'Numerical',
    hasEmbeddedAnswers: false,
    title: 'Sample Numerical Title',
    stem: { format: 'plain', text: 'Sample Stem' },
    choices: [
        { isCorrect: true, weight: 1, text: { type: 'simple', number: 42}, feedback: { format: 'plain', text: 'Correct!' } },
        { isCorrect: false, weight: 1, text: { type: 'simple', number: 43}, feedback: { format: 'plain', text: 'Incorrect!' } }
    ],
    globalFeedback: { format: 'plain', text: 'Sample Global Feedback' }
};

const htmlMock: TemplateOptions & NumericalType = {
    type: 'Numerical',
    hasEmbeddedAnswers: false,
    title: 'Sample Numerical Title',
    stem: { format: 'html', text: '$$\\frac{zzz}{yyy}$$' },
    choices: [
        { isCorrect: true, weight: 1, text: { type: 'simple', number: 42}, feedback: { format: 'html', text: 'Correct!' } },
        { isCorrect: false, weight: 1, text: { type: 'simple', number: 43}, feedback: { format: 'html', text: 'Incorrect!' } }
    ],
    globalFeedback: { format: 'html', text: 'Sample Global Feedback' }
};

const moodleMock: TemplateOptions & NumericalType = {
    type: 'Numerical',
    hasEmbeddedAnswers: false,
    title: 'Sample Numerical Title',
    stem: { format: 'moodle', text: 'Sample Stem' },
    choices: [
        { isCorrect: true, weight: 1, text: { type: 'simple', number: 42}, feedback: { format: 'moodle', text: 'Correct!' } },
        { isCorrect: false, weight: 1, text: { type: 'simple', number: 43}, feedback: { format: 'moodle', text: 'Incorrect!' } }
    ],
    globalFeedback: { format: 'moodle', text: 'Sample Global Feedback' }
};

const imageMock: TemplateOptions & NumericalType = {
    type: 'Numerical',
    hasEmbeddedAnswers: false,
    title: 'Sample Numerical Title with Image',
    stem: { format: 'plain', text: 'Sample Stem with Image' },
    choices: [
        { isCorrect: true, weight: 1, text: { type: 'simple', number: 42}, feedback: { format: 'plain', text: 'Correct!' } },
        { isCorrect: false, weight: 1, text: { type: 'simple', number: 43}, feedback: { format: 'plain', text: 'Incorrect!' } },
        { isCorrect: false, weight: 1, text: { type: 'simple', number: 44}, feedback: { format: 'plain', text: '<img src="https://via.placeholder.com/150" alt="Sample Image" />' } }
    ],
    globalFeedback: { format: 'plain', text: 'Sample Global Feedback with Image' }
};

test('Numerical snapshot test with plain text', () => {
    const { asFragment } = render(<Numerical {...plainTextMock} />);
    expect(asFragment()).toMatchSnapshot();
});

test('Numerical snapshot test with html', () => {
    const { asFragment } = render(<Numerical {...htmlMock} />);
    expect(asFragment()).toMatchSnapshot();
});

test('Numerical snapshot test with moodle', () => {
    const { asFragment } = render(<Numerical {...moodleMock} />);
    expect(asFragment()).toMatchSnapshot();
});

test('Numerical snapshot test with image', () => {
    const { asFragment } = render(<Numerical {...imageMock} />);
    expect(asFragment()).toMatchSnapshot();
});