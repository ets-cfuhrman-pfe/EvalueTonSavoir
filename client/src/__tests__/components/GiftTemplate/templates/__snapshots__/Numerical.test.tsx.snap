// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Numerical snapshot test with html 1`] = `
<DocumentFragment>
  &lt;section style="
  flex-wrap: wrap;
  position: relative;
  padding: 1rem 1rem;
  margin-bottom: 0.5rem;
  background-color: hsl(0, 0%, 100%);
  border: solid hsl(0, 0%, 100%) 2px;
  border-radius: 6px;
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
"&gt;
  &lt;div style="
  display: flex;
  font-weight: bold;
"&gt;
    &lt;span&gt;
      &lt;span style="
  color: #5271FF;
  "&gt;Sample Numerical Title&lt;/span&gt;
    &lt;/span&gt;
    &lt;span style="
  margin-left: auto;
  padding-left: 0.75rem;
  flex: none;
 margin-bottom: 10px;"&gt;
      &lt;span style="
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
  padding-left: 0.7rem;
  padding-right: 0.7rem;
  padding-top: 0.4rem;
  padding-bottom: 0.4rem;
  border-radius: 4px;
  background-color: hsl(0, 0%, 100%);
  color: hsl(180, 15%, 41%);
"&gt;Numérique&lt;/span&gt;
    &lt;/span&gt;
  &lt;/div&gt;
&lt;p style="
  color: hsl(0, 0%, 0%);
"&gt;&lt;span class="katex-display"&gt;&lt;span class="katex"&gt;&lt;span class="katex-mathml"&gt;&lt;math xmlns="http://www.w3.org/1998/Math/MathML" display="block"&gt;&lt;semantics&gt;&lt;mrow&gt;&lt;mfrac&gt;&lt;mrow&gt;&lt;mi&gt;z&lt;/mi&gt;&lt;mi&gt;z&lt;/mi&gt;&lt;mi&gt;z&lt;/mi&gt;&lt;/mrow&gt;&lt;mrow&gt;&lt;mi&gt;y&lt;/mi&gt;&lt;mi&gt;y&lt;/mi&gt;&lt;mi&gt;y&lt;/mi&gt;&lt;/mrow&gt;&lt;/mfrac&gt;&lt;/mrow&gt;&lt;annotation encoding="application/x-tex"&gt;\\frac{zzz}{yyy}&lt;/annotation&gt;&lt;/semantics&gt;&lt;/math&gt;&lt;/span&gt;&lt;span class="katex-html" aria-hidden="true"&gt;&lt;span class="base"&gt;&lt;span class="strut" style="height:1.988em;vertical-align:-0.8804em;"&gt;&lt;/span&gt;&lt;span class="mord"&gt;&lt;span class="mopen nulldelimiter"&gt;&lt;/span&gt;&lt;span class="mfrac"&gt;&lt;span class="vlist-t vlist-t2"&gt;&lt;span class="vlist-r"&gt;&lt;span class="vlist" style="height:1.1076em;"&gt;&lt;span style="top:-2.314em;"&gt;&lt;span class="pstrut" style="height:3em;"&gt;&lt;/span&gt;&lt;span class="mord"&gt;&lt;span class="mord mathnormal" style="margin-right:0.03588em;"&gt;yyy&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;span style="top:-3.23em;"&gt;&lt;span class="pstrut" style="height:3em;"&gt;&lt;/span&gt;&lt;span class="frac-line" style="border-bottom-width:0.04em;"&gt;&lt;/span&gt;&lt;/span&gt;&lt;span style="top:-3.677em;"&gt;&lt;span class="pstrut" style="height:3em;"&gt;&lt;/span&gt;&lt;span class="mord"&gt;&lt;span class="mord mathnormal" style="margin-right:0.04398em;"&gt;zzz&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;span class="vlist-s"&gt;​&lt;/span&gt;&lt;/span&gt;&lt;span class="vlist-r"&gt;&lt;span class="vlist" style="height:0.8804em;"&gt;&lt;span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;span class="mclose nulldelimiter"&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/span&gt;&lt;/p&gt;
      &lt;div&gt;
          &lt;span style="
  color: hsl(0, 0%, 0%);
"&gt;Réponse: &lt;/span&gt;&lt;input class="gift-input" type="text" style="
  display: inline-block;
  padding: 0.375rem 0.75rem;
  line-height: 1.5;
  color: hsl(0, 0%, 16%);
  background-color: hsl(0, 0%, 100%);
  border: 1px solid hsl(0, 0%, 81%);
  border-radius: 0.25rem;
  box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 2px 0px;
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  width: 90%;
" placeholder="42, 43"&gt;
      &lt;/div&gt;
      &lt;div style="
  position: relative;
  margin-top: 1rem;
  padding: 0 1rem;
  background-color: hsl(43, 100%, 94%);
  color: hsl(43, 95%, 9%);
  border: hsl(36, 84%, 93%) 1px solid;
  border-radius: 6px;
  box-shadow: 0px 2px 5px hsl(0, 0%, 74%);
"&gt;
        &lt;p&gt;Sample Global Feedback&lt;/p&gt;
      &lt;/div&gt;&lt;/section&gt;
</DocumentFragment>
`;

exports[`Numerical snapshot test with image 1`] = `
<DocumentFragment>
  &lt;section style="
  flex-wrap: wrap;
  position: relative;
  padding: 1rem 1rem;
  margin-bottom: 0.5rem;
  background-color: hsl(0, 0%, 100%);
  border: solid hsl(0, 0%, 100%) 2px;
  border-radius: 6px;
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
"&gt;
  &lt;div style="
  display: flex;
  font-weight: bold;
"&gt;
    &lt;span&gt;
      &lt;span style="
  color: #5271FF;
  "&gt;Sample Numerical Title with Image&lt;/span&gt;
    &lt;/span&gt;
    &lt;span style="
  margin-left: auto;
  padding-left: 0.75rem;
  flex: none;
 margin-bottom: 10px;"&gt;
      &lt;span style="
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
  padding-left: 0.7rem;
  padding-right: 0.7rem;
  padding-top: 0.4rem;
  padding-bottom: 0.4rem;
  border-radius: 4px;
  background-color: hsl(0, 0%, 100%);
  color: hsl(180, 15%, 41%);
"&gt;Numérique&lt;/span&gt;
    &lt;/span&gt;
  &lt;/div&gt;
&lt;p style="
  color: hsl(0, 0%, 0%);
"&gt;Sample Stem with Image&lt;/p&gt;
      &lt;div&gt;
          &lt;span style="
  color: hsl(0, 0%, 0%);
"&gt;Réponse: &lt;/span&gt;&lt;input class="gift-input" type="text" style="
  display: inline-block;
  padding: 0.375rem 0.75rem;
  line-height: 1.5;
  color: hsl(0, 0%, 16%);
  background-color: hsl(0, 0%, 100%);
  border: 1px solid hsl(0, 0%, 81%);
  border-radius: 0.25rem;
  box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 2px 0px;
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  width: 90%;
" placeholder="42, 43, 44"&gt;
      &lt;/div&gt;
      &lt;div style="
  position: relative;
  margin-top: 1rem;
  padding: 0 1rem;
  background-color: hsl(43, 100%, 94%);
  color: hsl(43, 95%, 9%);
  border: hsl(36, 84%, 93%) 1px solid;
  border-radius: 6px;
  box-shadow: 0px 2px 5px hsl(0, 0%, 74%);
"&gt;
        &lt;p&gt;Sample Global Feedback with Image&lt;/p&gt;
      &lt;/div&gt;&lt;/section&gt;
</DocumentFragment>
`;

exports[`Numerical snapshot test with moodle 1`] = `
<DocumentFragment>
  &lt;section style="
  flex-wrap: wrap;
  position: relative;
  padding: 1rem 1rem;
  margin-bottom: 0.5rem;
  background-color: hsl(0, 0%, 100%);
  border: solid hsl(0, 0%, 100%) 2px;
  border-radius: 6px;
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
"&gt;
  &lt;div style="
  display: flex;
  font-weight: bold;
"&gt;
    &lt;span&gt;
      &lt;span style="
  color: #5271FF;
  "&gt;Sample Numerical Title&lt;/span&gt;
    &lt;/span&gt;
    &lt;span style="
  margin-left: auto;
  padding-left: 0.75rem;
  flex: none;
 margin-bottom: 10px;"&gt;
      &lt;span style="
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
  padding-left: 0.7rem;
  padding-right: 0.7rem;
  padding-top: 0.4rem;
  padding-bottom: 0.4rem;
  border-radius: 4px;
  background-color: hsl(0, 0%, 100%);
  color: hsl(180, 15%, 41%);
"&gt;Numérique&lt;/span&gt;
    &lt;/span&gt;
  &lt;/div&gt;
&lt;p style="
  color: hsl(0, 0%, 0%);
"&gt;Sample Stem&lt;/p&gt;
      &lt;div&gt;
          &lt;span style="
  color: hsl(0, 0%, 0%);
"&gt;Réponse: &lt;/span&gt;&lt;input class="gift-input" type="text" style="
  display: inline-block;
  padding: 0.375rem 0.75rem;
  line-height: 1.5;
  color: hsl(0, 0%, 16%);
  background-color: hsl(0, 0%, 100%);
  border: 1px solid hsl(0, 0%, 81%);
  border-radius: 0.25rem;
  box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 2px 0px;
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  width: 90%;
" placeholder="42, 43"&gt;
      &lt;/div&gt;
      &lt;div style="
  position: relative;
  margin-top: 1rem;
  padding: 0 1rem;
  background-color: hsl(43, 100%, 94%);
  color: hsl(43, 95%, 9%);
  border: hsl(36, 84%, 93%) 1px solid;
  border-radius: 6px;
  box-shadow: 0px 2px 5px hsl(0, 0%, 74%);
"&gt;
        &lt;p&gt;Sample Global Feedback&lt;/p&gt;
      &lt;/div&gt;&lt;/section&gt;
</DocumentFragment>
`;

exports[`Numerical snapshot test with plain text 1`] = `
<DocumentFragment>
  &lt;section style="
  flex-wrap: wrap;
  position: relative;
  padding: 1rem 1rem;
  margin-bottom: 0.5rem;
  background-color: hsl(0, 0%, 100%);
  border: solid hsl(0, 0%, 100%) 2px;
  border-radius: 6px;
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
"&gt;
  &lt;div style="
  display: flex;
  font-weight: bold;
"&gt;
    &lt;span&gt;
      &lt;span style="
  color: #5271FF;
  "&gt;Sample Numerical Title&lt;/span&gt;
    &lt;/span&gt;
    &lt;span style="
  margin-left: auto;
  padding-left: 0.75rem;
  flex: none;
 margin-bottom: 10px;"&gt;
      &lt;span style="
  box-shadow: 0px 1px 3px hsl(0, 0%, 74%);
  padding-left: 0.7rem;
  padding-right: 0.7rem;
  padding-top: 0.4rem;
  padding-bottom: 0.4rem;
  border-radius: 4px;
  background-color: hsl(0, 0%, 100%);
  color: hsl(180, 15%, 41%);
"&gt;Numérique&lt;/span&gt;
    &lt;/span&gt;
  &lt;/div&gt;
&lt;p style="
  color: hsl(0, 0%, 0%);
"&gt;Sample Stem&lt;/p&gt;
      &lt;div&gt;
          &lt;span style="
  color: hsl(0, 0%, 0%);
"&gt;Réponse: &lt;/span&gt;&lt;input class="gift-input" type="text" style="
  display: inline-block;
  padding: 0.375rem 0.75rem;
  line-height: 1.5;
  color: hsl(0, 0%, 16%);
  background-color: hsl(0, 0%, 100%);
  border: 1px solid hsl(0, 0%, 81%);
  border-radius: 0.25rem;
  box-shadow: rgba(0, 0, 0, 0.05) 0px 1px 2px 0px;
  margin: 0;
  font-family: inherit;
  font-size: inherit;
  line-height: inherit;
  width: 90%;
" placeholder="42, 43"&gt;
      &lt;/div&gt;
      &lt;div style="
  position: relative;
  margin-top: 1rem;
  padding: 0 1rem;
  background-color: hsl(43, 100%, 94%);
  color: hsl(43, 95%, 9%);
  border: hsl(36, 84%, 93%) 1px solid;
  border-radius: 6px;
  box-shadow: 0px 2px 5px hsl(0, 0%, 74%);
"&gt;
        &lt;p&gt;Sample Global Feedback&lt;/p&gt;
      &lt;/div&gt;&lt;/section&gt;
</DocumentFragment>
`;
